services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/asn1/TimDecoder.java

```diff
@@ -0,0 +1,88 @@
+package us.dot.its.jpo.ode.api.asn1;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import us.dot.its.jpo.ode.api.models.messages.TimDecodedMessage;
+import us.dot.its.jpo.ode.api.models.messages.DecodedMessage;
+import us.dot.its.jpo.ode.api.models.messages.EncodedMessage;
+import us.dot.its.jpo.ode.model.Asn1Encoding;
+import us.dot.its.jpo.ode.model.Asn1Encoding.EncodingRule;
+import us.dot.its.jpo.ode.model.OdeAsn1Data;
+import us.dot.its.jpo.ode.model.OdeAsn1Payload;
+import us.dot.its.jpo.ode.model.OdeTimData;
+import us.dot.its.jpo.ode.model.OdeTimMetadata;
+import us.dot.its.jpo.ode.model.OdeData;
+import us.dot.its.jpo.ode.model.OdeHexByteArray;
+import us.dot.its.jpo.ode.model.OdeLogMetadata.RecordType;
+import us.dot.its.jpo.ode.model.OdeMsgMetadata.GeneratedBy;
+import us.dot.its.jpo.ode.model.OdeMsgPayload;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+@Slf4j
+@Component
+public class TimDecoder implements Decoder {
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public DecodedMessage decode(EncodedMessage message) {
+        
+        // Convert to Ode Data type and Add Metadata
+        OdeData data = getAsOdeData(message.getAsn1Message());
+
+        XmlUtils xmlUtils = new XmlUtils();
+
+        try {
+            // Convert to XML for ASN.1 Decoder
+            String xml = xmlUtils.toXml(data);
+
+            // Send String through ASN.1 Decoder to get Decoded XML Data
+            String decodedXml = DecoderManager.decodeXmlWithAcm(xml);
```

Comment:
chore(blocking): remove unused ObjectMapper, only instantiate XmlUtils once (it's not stateful)
```suggestion
    private final XmlUtils xmlUtils = new XmlUtils();

    @Override
    public DecodedMessage decode(EncodedMessage message) {
        
        // Convert to Ode Data type and Add Metadata
        OdeData data = getAsOdeData(message.getAsn1Message());
        
        try {
            // Convert to XML for ASN.1 Decoder
            String xml = xmlUtils.toXml(data);

            // Send String through ASN.1 Decoder to get Decoded XML Data
            String decodedXml = DecoderManager.decodeXmlWithAcm(xml);
```